lib/src/common/orgs_service.dart

```diff
@@ -167,6 +175,29 @@ class OrganizationsService extends Service {
     return completer.future;
   }
 
+  /// Returns the membership status for a user in a team given the [orgName] and [teamName].
+  Future<TeamMembershipState> getTeamMembershipByName(
+      String orgName, String teamName, String user) {
+    final completer = Completer<TeamMembershipState>();
+
+    github
```

Comment:
Is the completer necessary? You should be able to `return github.getJSON(...)` directly

Reply:
Yeah it was not necessary. In fact that whole fail on was not necessary since the fail function provided did not throw it did not even matter what it did so I removed it.